drivers/serial/uart_mcux.c

```diff
@@ -401,21 +403,30 @@ static const struct uart_mcux_config uart_mcux_##n##_config = {		\
 }
 
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+#define UART_MCUX_NONE
+
 #define UART_MCUX_CONFIG_FUNC(n)					\
 	static void uart_mcux_config_func_##n(const struct device *dev)	\
 	{								\
-		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(n, status, irq),	\
-			    DT_INST_IRQ_BY_NAME(n, status, priority),	\
-			    uart_mcux_isr, DEVICE_DT_INST_GET(n), 0);	\
-									\
-		irq_enable(DT_INST_IRQ_BY_NAME(n, status, irq));	\
-									\
-		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(n, error, irq),		\
-			    DT_INST_IRQ_BY_NAME(n, error, priority),	\
+		UART_MCUX_IRQ(n, status);	\
+		UART_MCUX_IRQ(n, error);	\
+	}
+
+#define UART_MCUX_IRQ_INIT(n, name)					\
+	do {								\
+		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(n, name, irq),	\
+			    DT_INST_IRQ_BY_NAME(n, name, priority),	\
 			    uart_mcux_isr, DEVICE_DT_INST_GET(n), 0);	\
 									\
-		irq_enable(DT_INST_IRQ_BY_NAME(n, error, irq));		\
-	}
+		irq_enable(DT_INST_IRQ_BY_NAME(n, name, irq));	\
+	} while (false)
+
+#define UART_MCUX_IRQ(n, name)						\
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(n, name),		\
+		    (UART_MCUX_IRQ_INIT(n, name)),			\
+		    (UART_MCUX_NONE))
```

Comment:
instead of defining UART_MCUX_NONE, I believe you could just leave the parentheses empty

Reply:
Yes, that sounds more concise